### ISSUE TITLE
Deserializing unknown Geometries?

### ISSUE THREAD

**Issue by FrankyBoy** [GeoJSON-Net/GeoJSON.Net#184]
Is there any way to just generically convert a Geometry into whatever type is specified in the GeoJSON itself? I.e. I have a DB column that might be anything from Point to MultiPolygon, so doing `JsonConvert.DeserializeObject<SomeSpecificGeomTypeHere>(data.geom)` does not work for obvious reasons. 

If there is, this should really be documented.

**Comment by xfischer** [comment]
Hi ! I think this will work : `JsonConvert.DeserializeObject<IGeometryObject>`

Good point about the documentation. Where would you like it to be documented? The code itself?

**Comment by xfischer** [comment]
You can also have a look at the tests here : https://github.com/GeoJSON-Net/GeoJSON.Net/blob/master/src/GeoJSON.Net.Tests/Geometry/GeometryTests.cs

**Comment by FrankyBoy** [comment]
Hi, so either `JsonConvert.DeserializeObject<IGeometryObject>(data.geom, new GeometryConverter())` or `JsonConvert.DeserializeObject<IGeoJsonObject>(data.geom, new GeoJsonConverter())` seemed to do the trick. To me it's unclear what the difference between those is as they both seem to produce the same result. 

Where it should be documented? I'd say the README as that seems to be the only up to date documentation / reference out there, and the converters are not shown at all there (or explanations what the different converters are meant for).

**Comment by xfischer** [comment]
Yeah I admit I also feel confused about that and wonder why it has been designed this way.

`IGeometryObject` is a minimal internal contract, the constraint of implementing this is *only* to be able to state which geometry type we are

`IGeoJsonObject` is more exhaustive, as it adds bounding box and CRS, it is a bridge made for adherence to GeoJSON specs IMHO.
